### PR TITLE
Remove Answer Text View in Url Widget

### PR DESCRIPTION
**Pull request by SaumiaSinghal**
This PR updates the UI of the URL widget. As discussed in ODK forum [here](https://forum.getodk.org/t/use-materialcardview-to-update-the-ui-of-url-widget/27500/6), the answer text view is removed and user simply sees the button to open the url in the widget. 

As discussed with @seadowg, this PR does not change the button text for the moment, so `Open Url` button is shown. As the user is not allowed to change the url, the widget is `readonly` all times, so I am removing any specific behavior concerning the `readonly` widget in this PR.

WhatsApp Image 2020-08-12 at 03 13 53

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
I changed the value fo the string resource `open_url` to `Visit Website`, and that change is not migrated to all the other languages of Collect, so this change won't appear good in when the user views the form in a language other than English. 

#### Do we need any specific form for testing your changes? If so, please attach one.
Any form having URL Widget like AllWidgetsForm.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

**Comment by seadowg** [comment]
> I still need to decide upon some issues regarding the behavior of URL widget. According to the current implementation, url button is not shown when the widget is readonly, but as the user is not allowed to change the answer anytime, so the widget is readonly at all times. I feel like not showing/disabling the url button when the url is empty.

My thinking here is that in the `readonly` case the widget should be no different as the "answer" is never changeable. Would be good to get second opinions from @grzesiek2010 and @lognaturel on that though.

**Comment by seadowg** [comment]
Another point to make here is that it's been really hard to make calls about the URL widget as we don't have a clear picture of how it is used or even if it is used at all. I think as part of this change it would be nice to add an analytics call that gets made when a URL widget is included in a form. 

We could probably have the `WidgetFactory` make a call whenever it creates a `URLWidget`. I'd imagine this could use the "Prompt" category (that audio labels are using) and pass the form analytics ID (`QuestionDetails#getFormAnalyticsID()`) to make sure we can answer the question "how many forms contain URL widgets?" rather than just "how many times is a URL widget rendered?". @SaumiaSinghal could you look at adding that in as part of these changes?

File: collect_app/src/test/java/org/odk/collect/android/widgets/UrlWidgetTest.java [hunk 1]
```diff
@@ -58,9 +59,9 @@ public void getAnswer_whenPromptHasAnswer_returnsAnswer() {
     }
 
     @Test
-    public void usingReadOnlyOption_makeAllClickableElementsDisabled() {
+    public void usingReadOnlyOption_doesNotShowUrlButton() {
         UrlWidget widget = createWidget(promptWithReadOnly());
-        assertThat(widget.getBinding().urlButton.getVisibility(), equalTo(View.GONE));
+        assertThat(widget.binding.urlButton.getVisibility(), equalTo(View.GONE));
```
Comment (seadowg) [inline collect_app/src/test/java/org/odk/collect/android/widgets/UrlWidgetTest.java]: As discussed the URL widget should behave exactly the same in read only mode (as it is always "read only"). That being the case you shouldn't need any tests for read only for the widget.

File: collect_app/src/main/java/org/odk/collect/android/widgets/UrlWidget.java [hunk 2]
```diff
@@ -44,7 +43,6 @@ public UrlWidget(Context context, QuestionDetails questionDetails, CustomTabHelp
     @Override
     protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
         binding = UrlWidgetAnswerBinding.inflate(((Activity) context).getLayoutInflater());
-        View answerView = binding.getRoot();
 
         if (prompt.isReadOnly()) {
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/widgets/UrlWidget.java]: As in other comments: lets remove any change in behaviour based on `isReadOnly`

File: collect_app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -217,7 +217,7 @@
     <string name="constraint_behavior_on_finalize">Defer validation until finalized</string>
     <string name="view_change_location">View or Change Location</string>
     <string name="change_location">Change Location</string>
-    <string name="open_url">Open Url</string>
+    <string name="open_url">Visit Website</string>
```
Comment (seadowg) [inline collect_app/src/main/res/values/strings.xml]: Lets not change this text at the moment.

**Review by seadowg — CHANGES_REQUESTED** [review]

**Review by seadowg — APPROVED** [review]

File: collect_app/src/main/java/org/odk/collect/android/analytics/AnalyticsEvents.java [hunk 4]
```diff
@@ -72,6 +72,12 @@ private AnalyticsEvents() {
      */
     public static final String AUDIO_QUESTION = "Prompt";
 
+    /**
+     * Track form definitions with url widget. The action should be the widget name and
+     * the label should be a hash of the form definition.
+     */
+    public static final String URL_WIDGET = "Prompt";
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/analytics/AnalyticsEvents.java]: This doesn't seem right as it's the same as the `AUDIO_QUESTION`. I think the `AUDIO_QUESTION` naming and description is just wrong here and probably should be `PROMPT` and really be used to just track any kind of question/widget.

File: collect_app/src/main/java/org/odk/collect/android/widgets/WidgetFactory.java [hunk 5]
```diff
@@ -122,6 +124,8 @@ public static QuestionWidget createWidgetFromPrompt(FormEntryPrompt prompt, Cont
                             questionWidget = new StringNumberWidget(context, questionDetails, readOnlyOverride);
                         } else if (appearance.equals(WidgetAppearanceUtils.URL)) {
                             questionWidget = new UrlWidget(context, questionDetails, new CustomTabHelper());
+
+                            analytics.logEvent(URL_WIDGET, "UrlWidget", questionDetails.getFormAnalyticsID());
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/widgets/WidgetFactory.java]: I think the action here should just be `"Url"`. What we care about is that the user saw a url question.

**Review by seadowg — CHANGES_REQUESTED** [review]
Just a couple of small changes to go in!

**Review by seadowg — APPROVED** [review]

**Comment by seadowg** [comment]
@lognaturel I think this would be nice to include in the release as it adds analytics. What do you think?

**Comment by mmarciniak90** [comment]
@SaumiaSinghal please take a look at my test results. 



>I changed the value fo the string resource open_url to Visit Website

:red_circle: **Visit Website** text on the button is not visible

> I feel like not showing/disabling the url button when the url is empty.

:red_circle: I added a screenshot where url is empty and as you can see, button is visible and it's clickable. Toast with message appears

:large_blue_circle:  website address is not visible under the button -> the result is that the user has no idea whether the URL is set or not until the user clicks on button or open hierarchy view. It seems ok but it is worth paying attention to.
:green_circle: remove reponse dialog is not available for readonly url widget
:green_circle: readonly url widget has button and visit website is possible

**Comment by SaumiaSinghal** [comment]
Hi @mmarciniak90! Thanks for testing the PR. I am sorry, I forgot to change the description of this PR. We decided not to change the button text for the moment, as we have to look into the string translations then. Also, we will not be showing the answer textview of the widget as mentioned in the header of the PR. Regarding the case when there is no answer set, I'll confirm about that case with my mentor and update the PR. Thanks!

**Comment by seadowg** [comment]
I think for the no answer case we can just go with the current behaviour (that @mmarciniak90 is seeing). I think we're very unlikely to see that case in real life as it would be a mistake in the form and the current behaviour is actually pretty good for bubbling that mistake up to the user without interrupting the form.

**Comment by kkrawczyk123** [comment]
**Tested with success!**
Verified on Androids: 5.1, 7.0, 7.1, 8.1, 9.0 and 10

Tested cases:
- light/dark mode
- read only
- open url
- minimize app
- screen rotation
- RTL language 
- try to remove answer

@getodk-bot unlabel "needs testing"
@getodk-bot label "behavior verified"

**Comment by kkrawczyk123** [comment]
@getodk-bot label "behavior verified"

**Comment by seadowg** [comment]
@SaumiaSinghal looks like we've got conflicts and failing checks.

**Comment by SaumiaSinghal** [comment]
Done!

**Pull request merged**